cmd/clusterctl/client/cluster/components_test.go

```diff
@@ -83,6 +88,8 @@ func Test_providerComponents_Delete(t *testing.T) {
 		},
 		// CRDs (should be deleted only if includeCRD)
 		&crd,
+		// Inventory should be deleted ony if includeInventory
+		&inventory,
```

Comment:
There's an ambiguity between the 'provider' set in this test and the 'inventory' that this PR actually addresses. I chose the less disruptive option here of just adding the 'Provider' type under the name inventory, and using the name inventory through the code for this PR. Should the ambiguity between the inventory provider type and the use of provider elsewhere in the code be addressed here?

Reply:
what about using IncludeInventory instead of inventory?

Reply:
That's better than what I have here, but there's a broader issue that we have two types - a Go Provider type that stands in for providers in the code, and an api CRD  'Provider.clusterctl' type that serves as the inventory for clusterctl. Is this enough ambiguity to consider 1) Merging these two in the code or 2) Renaming the Provider api type to something that reflects its usage by clusterctl?

Reply:
Updated to providerInventory for now. I think that solves the ambiguity in the test, but not in other places where this issue might come up.